internal/dangling/dangling_test.go

```diff
@@ -0,0 +1,208 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dangling
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	nropv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1"
+	nodegroupv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1/helper/nodegroup"
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+const (
+	nrouid  = "123"
+	ns      = "nro-ns"
+	nroName = "nro"
+)
+
+var (
+	targetMCPName = "mcp1" // var because we need its pointer
+	ctx           = context.TODO()
+
+	nro = nropv1.NUMAResourcesOperator{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nroName,
+			Namespace: ns,
+			UID:       nrouid,
+		},
+	}
+
+	trees = []nodegroupv1.Tree{
+		{
+			NodeGroup: &nropv1.NodeGroup{
+				PoolName: &targetMCPName,
+			},
+			MachineConfigPools: []*mcov1.MachineConfigPool{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: targetMCPName,
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestDeleteUnusedMachineConfigs(t *testing.T) {
+	mcs := []mcov1.MachineConfig{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("51-%s-%s", nroName, targetMCPName),
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						UID: nrouid,
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("51-%s-mcp2", nroName),
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						UID: nrouid, // owned but dangling
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "51-XX-mcp1",
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						UID: "100", // not owned by the NRO object
+					},
+				},
+			},
+		},
+	}
+
+	_ = mcov1.AddToScheme(scheme.Scheme)
+	fakeClient := fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(&mcs[0], &mcs[1], &mcs[2]).Build()
+
+	err := isMachineConfigFound(fakeClient, ctx, mcs...)
+	if err != nil {
+		t.Errorf("failed to find MachineConfigs: %v", err)
+		return
+	}
+
+	err = DeleteUnusedMachineConfigs(fakeClient, context.TODO(), &nro, trees)
+	if err != nil {
+		t.Fatalf("failed to delete unused dangling machine configs: %v", err)
+		return
+	}
+
+	err = isMachineConfigFound(fakeClient, context.TODO(), mcs[0], mcs[2])
+	if err != nil {
+		t.Errorf("failed to find owned non-dangling MachineConfigs: %v", err)
+	}
+	err = isMachineConfigFound(fakeClient, context.TODO(), mcs[1])
+	if err == nil {
+		t.Errorf("succeeded to get dangling mc1 from fake client while expected to be deleted")
```

Comment:
Sorry for being pedantic but the expected flow here is to get an error so I wouldn't use the work "succeeded" to convey unexpected behavior. Also, mc1 in the error message is not the correct name right? 

We should either make it generic `Found dangling machine config from fake client which was expected to be deleted`. I think that a better way would be  mention its name  t.Errorf("Found dangling machine config: %s from fake client which was expected to be deleted", mcs[1].Name)

Reply:
Thanks for your comment. I agree, updating right away.